### DISCUSSION TITLE
[BUG] Transformer batch from csv  add [] to variable

### DISCUSSION THREAD

**Discussion by postheimer** [nextgenhealthcare/connect#4882] [General]
Hello 

i have a channel , he read a csv on process batch 

each row/column is mapper to a variable from iterator , all work goo but 

when i write the final flat file , all my variable have a [ ]   ? 

i'm in mirth connect 3.10.0  

i have tried a regex but no luck

**Comment by pacmano1** [comment]
Post your code and or the full channel.   We can't guess what you wrote.

**Comment by postheimer** [comment]


**Reply by pacmano1** [reply]
Moved your submission to the discussion area.   It is not a bug.  It looks like you are collecting all the csv rows into an array and then not iterating over the array in your destination.

**Reply by postheimer** [reply]
the final flat file is good but i have a result like this
,[identificationnumber],[patnumber],[dob],

i want only remove the []

:-)

**Reply by pacmano1** [reply]
Can you post a full screen shot of that please?

**Reply by postheimer** [reply]
![temp](https://user-images.githubusercontent.com/54264707/144278694-64f3b51f-896a-4558-b8e7-28f5cc4cf981.png)

**Reply by postheimer** [reply]
i have found a solution :-)

i send all my variable to channel writer to a new channel , and all [ ] as remove  👍

**Comment by zanajashanica** [comment]
Hi @postheimer, @pacmano1 , 

I have a channel that reads the CSV files, from Source File Reader to Destination File Writer.

I want to store some values of the file, using a mapper in the transformer. I don't know how can I access the file that I am receiving in destionation.

Can you help me?

Thanks, 
Zana

**Reply by pacmano1** [reply]
There are examples over on the mirth forums site and youtube that explain how to consume and map files.  This is a fundamental function of mirth.
